core/src/main/java/org/mule/runtime/core/privileged/exception/TemplateOnErrorHandler.java

```diff
@@ -76,6 +91,29 @@
   protected String errorType = null;
   protected ErrorTypeMatcher errorTypeMatcher = null;
 
+  private final Supplier<FluxSink<CoreEvent>> routingSink = new RoundRobinFluxSinkSupplier<>(getRuntime().availableProcessors(),
```

Comment:
Use an approach similar to https://github.com/mulesoft/mule/pull/7940#pullrequestreview-245036622 for when there are transactions